platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -900,6 +924,10 @@ - (id)mgl_jsonExpressionObject {
                 }
                 [NSException raise:NSInvalidArgumentException
                             format:@"Casting expression to %@ not yet implemented.", type];
+            }  else if ([function isEqualToString:@"mgl_expression:betweenLeftHandExpression:rightHandExpression:"] ||
+                        [function isEqualToString:@"mgl_between:"]) {
```

Comment:
Instead of adding a case here, add a case to `-[NSComparisonPredicate(MGLAdditions) mgl_jsonExpressionObject]` for `NSBetweenPredicateOperatorType`.